src/main/java/org/apache/mesos/hdfs/executor/NameNodeExecutor.java

```diff
@@ -106,13 +113,53 @@ public void frameworkMessage(ExecutorDriver driver, byte[] msg) {
       } else {
         deleteFile(nameDir);
         nameDir.mkdirs();
-        runCommand(driver, nameNodeTask, "bin/hdfs-mesos-namenode " + messageStr);
-        startProcess(driver, nameNodeTask);
-        startProcess(driver, zkfcNodeTask);
+      }
+      runCommand(driver, nameNodeTask, "bin/hdfs-mesos-namenode " + messageStr);
```

Comment:
I think this runCommand(init/bootstrap) does belong up in the else case, since we don't want to reformat the NN metadata if it already exists. The rest of the startProcess(NN + ZKFC) and sendStatusUpdate(RUNNING) lines probably belong outside of the if/else as you have them.